text/0000-inferred-type-aliases.md

```diff
@@ -0,0 +1,696 @@
+as exposing all operations the generated code affords:
+
+```rust
+type Strategy = _;
+```
+
+Simpler macros, including of the `macro_rules!` flavor,
+can similarly be simplified and require less input from the user by using `_`.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## Vocabulary
+
+First, let's get some useful vocabulary out of the way.
+
+### `inferred`
+[inferred]: #inferred
+
+[type_inference]: https://en.wikipedia.org/wiki/Type_inference
+
+By [*"inferred"*][type_inference] we refer to the compiler automatically
+determining type of some object such as an expression for you.
+
+For example, when you write:
+
+```rust
+let yummy_food = "Tikka masala";
+
+// or equivalently:
+
+let yummy_food: _ = "Tikka masala";
+```
+
+the compiler will automatically infer that the type of `yummy_food`
+is `&'static str` here.
+
+### `transparent`
+[transparent]: #transparent
+
+For something to be *"transparent"* means that you can see through it.
+
+In the context of Rust and in this RFC in particular we mean that for the
+`type` of some object / expression to be transparent means that its true
+nominal type can be obtained by the user. This entails that if you have
+some expression of that type, you can perform all operations the type affords.
+
+An example:
+
+```rust
+let foo: _     = make_some_number();
+// The compiler permit this because the type of `foo` is transparent:
+let bar: usize = foo;
+```
+
+### `opaque`
+[opaque]: #opaque
+
+The opposite of something being transparent is for something to be *"opaque"*.
+This means that you can *not* see through it. In the context of Rust,
+this refers to that the true nominal type of some object or expression
+is hidden from the user. Instead only a particular interface is exposed.
+This entails that if you have such an expression, only the operations
+afforded by the interface may be performed. This has been provided by
+the `impl Trait` (see [RFC 1522], [RFC 1951], [RFC 2071], [RFC 2250],
+and currently proposed [RFC 2515]) mechanism in Rust.
+
+An example:
+
+```rust
+use std::ops::Range;
+
+fn one_to_hundred() -> impl Iterator<Item = u8> {
+    let range: Range<u8> = 0..100;
+    range
+}
+
+fn main() {
+    let _numbers: Range<u8> = one_to_hundred();
+}
+```
+
+This results in:
+
+```rust
+error[E0308]: mismatched types
+ --> src/main.rs:9:30
+  |
+9 |     let _numbers: Range<u8> = one_to_hundred();
+  |                               ^^^^^^^^^^^^^^^^ expected struct `std::ops::Range` found anonymized type
+  |
+  = note: expected type `std::ops::Range<u8>`
+             found type `impl std::iter::Iterator`
+
+error: aborting due to previous error
+```
+
+Here *"anonymized"* and *"opaque"* mean the same thing.
+As we can see from the error message, even though we know that the nominal type
+of `_numbers` is `Range<u8>`, the compiler has prevented us from concluding this.
+
+### `voldemort type`
```

Comment:
Bike-shedding, but I would personally prefer to avoid popular culture references, as the meaning is entirely lost on those unfamiliar with it (and I expect the proportion of people familiar with D is even smaller). "Unnameable type" seems ideal.

Reply:
Fixed :)